### PR TITLE
Skip properties and fields that cannot be set.

### LINKED ISSUE #2119: StackOverflowException with EncryptionMessageMutator
Hello,

I got that one when serializing this entity:

```c#
public class BogusEntity
{
    public int Id { get; set; }
    public byte[] Data { get; set; }

    //This property generates a stackoverflow
    public List<BogusEntity> Graphics
    {
        get
        {
            return new List<BogusEntity> { new BogusEntity { Id = Id, Data = Data} };
        }
    }
}
```

inside this message:
```c#
public interface IBogusEntityMessage : IMessage
{
    BogusEntity Entity { get; set; }
}
```

and sent this way:
```c#
Bus.Send<IBogusEntityMessage>("StackOverflowRepro", message =>
{
    message.Entity = new BogusEntity
    {
        Id = 1,
        Data = new byte[] { 1, 2, 3, 6, 5, 4, 7, 8, 9, 5, 2, 1, 4, 5, 2, 1, 4, 5, 8, 7, 4, 
                5, 2, 1, 4, 5, 6, 8, 7, 4, 21, 63, 6, 8, 4, 4, 5, 69, 69, 7 }
    };
});
```

The 'Graphics' property is causing this. I also wonder why it is trying to transform a property with no setter.

### PR DESCRIPTION
**Pull request by johnsimons**
This PR fixes #2119

### CODE REVIEW

**Comment by johnsimons** [comment]
@Particular/nservicebus please review

File: src/NServiceBus.Core.Tests/Encryption/skip_props_and_fields_that_cannot_be_set.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿namespace NServiceBus.Core.Tests.Encryption
+{
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class skip_props_and_fields_that_cannot_be_set : WireEncryptedStringContext
```
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/Encryption/skip_props_and_fields_that_cannot_be_set.cs]: Nitpicking but if we go with "BDD" style syntaxes for unit tests I would name this:

```
class When_message_contains_props_and_fields_that_cannot_be_set
{
         [Test]
          public void Should_ignore_those_properties_and_fields()...
}
```

or something along those lines

**Comment by SimonCropp** [comment]
looks good to me

File: src/NServiceBus.Core.Tests/Encryption/skip_props_and_fields_that_cannot_be_set.cs (hunk 1 above)
Comment (johnsimons) [inline src/NServiceBus.Core.Tests/Encryption/skip_props_and_fields_that_cannot_be_set.cs]: thanks @andreasohlund, a lot better

**Pull request merged**